docs/topics/custom-vnet.md

```diff
@@ -81,4 +81,45 @@ spec:
 
 If no CIDR block is provided, `10.0.0.0/8` will be used by default, with default internal LB private IP `10.0.0.100`.
 
+### Ingress Rules
+
+Ingress rules can also be added as part of the subnet specification in a custom network spec.
+Note that ingress rules for the Kubernetes API Server port (default 6443) and SSH (22) are automatically added to the controlplane subnet.
```

Comment:
Should we be explicit about these being "additional ingress rules"? These don't replace the defaults, they are appended to the defaults. 

On the other hand, would it be valuable to instead allow the user to specify _all_ rules? I could see use cases for someone to want to disable SSH, so would it makes sense to instead of appending the rules have the webhook default the "default" rules if no rules are specified, but keep the user defined rules as-is if they are provided.

Your experience as a user of CAPZ is valuable here @rsmitty. As a user, would you prefer having full control over what rules are added (but have the burden of defining the right rules for API server and risk misconfiguring the rules which would result in a nonfunctional cluster), or have CAPZ implement the right required rules for you and only be able to add additional rules specific to your use case (but not remove rules).

Additional thoughts:
- If we do allow for "additional rules", could be good to either 1) make that explicit in the spec naming/description or 2) have the defaults merged into the list of rules in the AzureCluster webhook.
- Right now I'm leaning for giving the user full control with the assumption that if the user decides to customize rules they should know what they are doing, with pitfalls clearly documented. If the user just wants CAPZ to work they can always not define rules.

Reply:
Yeah, I think what you mention makes sense. The defaults should be "batteries included" with easy ways to get to the apiserver and ssh b/c they are set by default. If someone wants to do anything different, they can easily specify the same defaults along with their additional ingresses if need be.